plugins/modules/public_cloud_flavorid.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from __future__ import (absolute_import, division, print_function)
+
+from ansible.module_utils.basic import AnsibleModule
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: public_cloud_flavorid
+short_description: Get flavor id based on commercial name
+description:
+    - Get flavorId based on commercial name (t1-45, b2-7 etc)
+    - The flavorId change between region
+    - The retrieved flavorId can be used to spawn a new instance
+author: Synthesio SRE Team
+requirements:
+    - ovh >= 0.5.0
+options:
+    name:
+        required: true
+        description: The commercial name of the flavor (t1-45, b2-7 etc)
+    region:
+        required: true
+        description: The region where to lookup for flavorId
+    service_name:
+        required: true
+        description: The service_name
+
+'''
+
+EXAMPLES = '''
+- name: run installation
+  synthesio.ovh.public_cloud_flavorid
+    service_name: "{{ service_name }}"
+    region: "GRA7"
+    name: "t1-45"
+  delegate_to: localhost
+  register: flavor_id
+'''
+
+RETURN = ''' # '''
+
+from ansible_collections.synthesio.ovh.plugins.module_utils.ovh import ovh_api_connect, ovh_argument_spec
+
+try:
+    from ovh.exceptions import APIError
+    HAS_OVH = True
+except ImportError:
+    HAS_OVH = False
+
+
+def run_module():
+    module_args = ovh_argument_spec()
+    module_args.update(dict(
+        service_name=dict(required=True),
+        name=dict(required=True),
+        region=dict(required=True)
+    ))
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+    client = ovh_api_connect(module)
+
+    service_name = module.params['service_name']
+    name = module.params['name']
+    region = module.params['region']
+
+    try:
+        flavor = None
+        result = client.get('/cloud/project/%s/flavor' % (service_name),
+                            region="GRA7"
```

Comment:
```suggestion
                            region=region
```
Hardcoded value for testing forgotten